CMakeLists.txt

```diff
@@ -0,0 +1,251 @@
+cmake_minimum_required(VERSION 3.13)
+project(Gears_Vk)
+
+set(CMAKE_CXX_STANDARD 20)
+
+include(FetchContent)
+
+# ----------------------- Options -------------------------
+
+set(gvk_AllowedLibraryTypes INTERFACE SHARED STATIC)
```

Comment:
The reason I prefixed all options is to not accidentally override options in another project (a dependency or a dependent project of *Gears-Vk*).
For example, the CMake scripts of *ASSIMP* and *GLFW* both have an option called `BUILD_SHARED_LIBS` with different defaults. This is annoying when you want to have different behaviours for different libraries and I wanted to prevent such situations for future users by prefixing all options used by *Gears-Vk*.
Would you prefer a different prefix?